src/libsyntax/tokenstream.rs

```diff
@@ -182,6 +182,31 @@ pub struct TokenStream {
     kind: TokenStreamKind,
 }
 
+impl TokenStream {
+    /// Given a `TokenStream` with a `Stream` of only two arguments, return a new `TokenStream`
+    /// separating the two arguments with a comma for diagnostic suggestions.
+    pub(crate) fn add_comma(&self) -> Option<(TokenStream, Span)> {
+        // Used ot suggest if a user writes `println!("{}" a);`
+        if let TokenStreamKind::Stream(ref slice) = self.kind {
+            if slice.len() == 2 {
```

Comment:
So this doesn't work for `println!("{} {}" a, b);` or `println!("{} {}", a b);`?

If it does work, please add tests, if it does not work, would it be a lot of effort to fix that?

Reply:
The further we go from the `"{}" a` case, the less likely we're to find a suitable fix by just checking against comma separated list. Having said that, making sure that every element is separated by commas should be easy to add.